sql/parser/builtins.go

```diff
@@ -436,7 +436,11 @@ var builtins = map[string][]builtin{
 			types:      typeList{},
 			returnType: DummyDate,
 			fn: func(e EvalContext, args DTuple) (Datum, error) {
-				return MakeDDate(e.StmtTimestamp.Time), nil
+				loc, err := e.GetLocation()
+				if err != nil {
+					return DummyTimestamp, err
+				}
+				return MakeDDate(e.StmtTimestamp.Time.In(loc)), nil
```

Comment:
The only reason I've done it this way is to lazy evaluate. I think it's
okay given that it's confined to these built-ins

On Tue, Oct 6, 2015, 12:12 AM Ben Darnell <notifications@github.com> wrote:

> In sql/parser/builtins.go
> <https://github.com/cockroachdb/cockroach/pull/2774#discussion_r41224473>:
>
> > @@ -436,7 +436,11 @@ var builtins = map[string][]builtin{
> >  			types:      typeList{},
> >  			returnType: DummyDate,
> >  			fn: func(e EvalContext, args DTuple) (Datum, error) {
> > -				return MakeDDate(e.StmtTimestamp.Time), nil
> > +				loc, err := e.GetLocation()
> > +				if err != nil {
> > +					return DummyTimestamp, err
> > +				}
> > +				return MakeDDate(e.StmtTimestamp.Time.In(loc)), nil
>
> I think it would be better to set the appropriate location when we
> initialize e.StmtTimestamp. In general the server should never touch a
> timestamp whose location is time.Local. It should always be either UTC or
> the session location.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2774/files#r41224473>.
>